src/ripple/protocol/impl/STLedgerEntry.cpp

```diff
@@ -76,15 +76,19 @@ void STLedgerEntry::setSLEType ()
         Throw<std::runtime_error> ("invalid ledger entry type");
 
     type_ = mFormat->getType ();
-    if (!setType (mFormat->elements))
+    try
     {
-        if (auto j = debugJournal().warn())
+        setType(mFormat->elements);
+    }
+    catch(std::exception&)
```

Comment:
Just to be clear, the existing code throws exceptions on invalid inputs. But I can change that.